### PR TITLE
Issue/16

### LINKED ISSUE #16: Blocking Request
We should just allow people to specify on the request that they want the server to wait for the request to be done before it responds. This is very RPC-like, but for requests that take a small amount of time to execute, it makes more sense than polling.

For Example:

`curl -X POST http://localhost:2337/api/v1/requests?blocking=True $JSON_PAYLOAD`

### PR DESCRIPTION
**Pull request by hazmat345**
This fixes #16

### CODE REVIEW

**Comment by loganasherjones** [comment]
I'd like to see some tests for this. Either in the integration tests or unit tests.

**Comment by hazmat345** [comment]
Will add tests to the integration suite.

**Comment by hazmat345** [comment]
Added a maximum wait time that defaults to 30 seconds. That removes the blocking forever problem.

The issue with CANCELLED would be addressed if we used the REST API to update the status instead of just modifying the database. Not sure if that's a good idea since we'd generate a lot of extra traffic if there were a bunch of requests in the queue when we cleared it.

It would also be fixed if we combined brew-view and bartender - we could just notify the wait condition where we set the CANCELLED status.

What do you think?

File: brew_view/base_handler.py [hunk 1]
```diff
@@ -127,12 +128,18 @@ def write_error(self, status_code, **kwargs):
                 code = error_dict.get('status_code', 500)
                 message = error_dict.get('message', str(e))
 
+            # HTTPError is a special case - can be raised by Tornado itself
+            elif isinstance(e, HTTPError):
```
Comment (loganasherjones) [inline brew_view/base_handler.py]: Are we capable of raising a new error now? It seems like maybe we were just missing this case before?

Comment (hazmat345) [reply brew_view/base_handler.py]: Oh yeah, I'll pull this back out. I kind of found it by accident. I think we should still do this, but it doesn't need to be in this merge.

**Pull request merged**